### DISCUSSION TITLE
Just curious about Termux user IDs.

### DISCUSSION THREAD

**Discussion by macerg** [termux/termux-packages#9418] [Q&A] [answered]
Anyone know how these are allotted within Termux? I have two identical phones, on one the user ID is u0_a246@localhost, on the other it is u0_a247@localhost.
My phone is rooted and I see tasks running under many different u0_* user IDs. Maybe these are some sort of 'virtual users created for 'non-sytem' tasks? Just a guess.

**Comment by agnostic-apollo** [comment, marked as answer]
These are unique user ids assigned to each app per profile at install time by android so that apps run with their own specific capabilities/access and sandbox. Termux does not allocate them and you normally can't change them, other than possible root hacks.

```
# AID_USER_OFFSET 100000 /* offset for uid ranges for each user */
# AID_APP_START 10000 /* first app user */
# AID_APP_END 19999   /* last app user */
# (userid * AID_USER_OFFSET + AID_APP_START + appid)
# profile user 0: (0 * 100000 + 10000 + 400) -> 10400/u0_a400
# profile user 11: (11 * 100000 + 10000 + 400) -> 1110400/u11_a400
```
https://cs.android.com/android/platform/superproject/+/android-11.0.0_r40:system/core/libcutils/include/private/android_filesystem_config.h;l=189
https://cs.android.com/android/platform/superproject/+/android-11.0.0_r40:bionic/libc/bionic/grp_pwd.cpp;l=261
